torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -705,6 +706,12 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     }
 
     if (!print_inline_) {
+      if (op_type == UnaryOpType::RandLike) {
+        indent() << "Philox rnd" << uop->name() << "(philox_args.seed_, "
+                 << genTensorIndex(uop->in()->as<kir::TensorIndex>())
```

Comment:
I think this does not work if the input of randlike is a discontinuous tensor?
```C++
TEST_F(NVFuserTest, TMP) {
  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
  auto fusion = fusion_ptr.get();
  FusionGuard fg(fusion);

  TensorView* tv0 = makeConcreteTensor({10, 17});
  fusion->addInput(tv0);
  auto tv1 = randlike(tv0);
  fusion->addOutput(tv1);

  tv0->computeAt(tv1, -1);

  fusion->printKernel();

  const auto options =
      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
  auto input = at::zeros({10, 1000000}, options).narrow(1, 13, 17);

  FusionExecutor fe;
  fe.compileFusion(fusion, {input});
  auto outputs = fe.runFusion({input});

  std::cout << outputs[0] << "\n";
}
```
```CUDA
__global__ void CUDAGeneratedKernel(Tensor<float, 2> T0, Tensor<double, 2> T3, at::PhiloxCudaState philox_args) {
  const auto idx = ((((blockIdx.z * gridDim.y + blockIdx.y) * gridDim.x + blockIdx.x) * blockDim.z + threadIdx.z) * blockDim.y + threadIdx.y) * blockDim.x + threadIdx.x;  auto offset = philox_args.captured_ ?
    static_cast<uint64_t>(*(philox_args.offset_.ptr) + philox_args.offset_intragraph_) :
    philox_args.offset_.val;
  Philox rnd(philox_args.seed_, idx, offset);
  NVFUSER_DEFINE_MAGIC_ZERO
  #pragma unroll
  for(nvfuser_index_t i19 = 0; i19 < 10; ++i19) {
    int64_t i42;
    i42 = i19 + nvfuser_zero;
    #pragma unroll
    for(nvfuser_index_t i20 = 0; i20 < 17; ++i20) {
      int64_t i24;
      i24 = i20 + nvfuser_zero;
      float T1[1];
      T1[0] = 0;
      Philox rnd22(philox_args.seed_, (i19 * T0.stride[0]) + (i24 * T0.stride[1]), offset);
      T1[0]
         = randLikef(rnd22);
      bool T2[1];
      T2[0]
        = T1[0]
        == (float) 1.00000000000000000e+00;
      T3[(i19 * 17) + i24]
         = where(T2[0], (float) 0.00000000000000000e+00, T1[0]);
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO
}
```

For this case, you will not get linear index, so it can be hard to predict the offset you should increment for the next tensor.

But I think it will be fine for schedulers, because they always insert caches for fusion inputs, and the cache should be contiguous.

Reply:
Good point! So would need to ignore contiguity in this case as this is not really indexing a tensor.